qemu/tests/numa_cpu.py

```diff
@@ -0,0 +1,142 @@
+import logging
+
+from virttest import error_context
+from virttest.utils_misc import NumaInfo
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Assign cpu to numa node with "-numa cpu", check the numa info in monitor
+    and guest os match with the qemu cli
+    """
+
+    def convert_cpu_topology_to_index(socket, die, core, thread):
+        """
+        Convert the cpu topology to cpu index list
+        """
+        def _get_boundary(value, max_value, weight):
+            min_boundary = int(value if value is not None else 0) * weight
+            max_boundary = int(value if value is not None else (max_value - 1)) * weight
+            return (min_boundary, max_boundary)
+
+        if vm_arch in ('x86_64', 'i386'):
+            socket_min, socket_max = _get_boundary(socket, vcpu_sockets, socket_weight)
+            die_min, die_max = _get_boundary(die, vcpu_dies, die_weight)
+            core_min, core_max = _get_boundary(core, vcpu_cores, core_weight)
+            thread_min, thread_max = _get_boundary(thread, vcpu_threads, thread_weight)
+            cpu_min = socket_min + die_min + core_min + thread_min
+            cpu_max = socket_max + die_max + core_max + thread_max
+        elif vm_arch in ('ppc64', 'ppc64le'):
+            cpu_min = int(core)
+            cpu_max = int(core) + vcpu_threads - 1
+        cpu_index = range(cpu_min, cpu_max+1, 1)
+        return cpu_index
+
+    def numa_cpu_guest():
+        """
+        Get the numa cpus for each node in guest os, linux only
+        """
+        error_context.context("Get cpus in guest os", logging.info)
+        numa_info_guest = NumaInfo(session=session)
+        nodes_guest = numa_info_guest.online_nodes
+        numa_cpu_guest = []
+        for node in nodes_guest:
+            numa_cpus = numa_info_guest.online_nodes_cpus[node]
+            numa_cpus = set([int(v) for v in numa_cpus.split()])
+            numa_cpu_guest.append(numa_cpus)
+        return numa_cpu_guest
+
+    def numa_cpu_cli():
+        """
+        Get the numa cpus for each node according to the qemu cli
+        """
+        error_context.context("Get the expected cpus in qemu command line", logging.info)
+        cpus = params.objects("guest_numa_cpus")
+        numa_cpu_cli = []
+        tmp = {}
+        for cpu in cpus:
+            cpu_params = params.object_params(cpu)
+            nodeid = cpu_params["numa_cpu_nodeid"]
+            socket = cpu_params.get("numa_cpu_socketid")
```

Comment:
How about using socketid, dieid, coreid, threadid to be consistent with nodeid?

Reply:
> How about using socketid, dieid, coreid, threadid to be consistent with nodeid?

It is just a temporary param used in the following L65, to avoid wrap lines of L65, please allow me keep it, thanks~